third_party/blink/renderer/platform/loader/fetch/memory_cache.h

```diff
@@ -51,16 +51,32 @@ class KURL;
 // when the prefinalizer is executed.
 class MemoryCacheEntry final : public GarbageCollected<MemoryCacheEntry> {
  public:
-  explicit MemoryCacheEntry(Resource* resource) : resource_(resource) {}
+  explicit MemoryCacheEntry(Resource* resource) {
+    if (recordreplay::IsRecordingOrReplaying("leak-references")) {
+      leaked_resource_ = resource;
+    } else {
+      resource_ = resource;
+    }
+  }
 
   void Trace(Visitor*) const;
-  Resource* GetResource() const { return resource_; }
+  Resource* GetResource() const { 
+    if (recordreplay::IsRecordingOrReplaying("leak-references")) {
+      return leaked_resource_; 
+    } else {
+      return resource_; 
+    }
+  }
 
  private:
   void ClearResourceWeak(const LivenessBroker&);
 
   // We use UntracedMember<> here to do custom weak processing.
   UntracedMember<Resource> resource_;
+  
+  // Replay uses Member<> here to leak resources to avoid non-deterministic
+  // GC of resources.
+  Member<Resource> leaked_resource_;
```

Comment:
strong_resource_ might be a better name here (and in related comments) -- we're holding a strong reference on the resource, but when the cache is destroyed or the entry is evicted the resource can still be collected.  (Using "leak-references" for the feature is fine though, from one perspective holding onto the resource longer than necessary constitutes a leak.)